### PR TITLE
Use tell not asked to all optimizers in Portfolio

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

Not sure this is good, this is for discussion.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/optimizerlib.py [hunk 1]
```diff
@@ -903,10 +903,23 @@ def _internal_ask_candidate(self) -> p.Parameter:
 
     def _internal_tell_candidate(self, candidate: p.Parameter, value: float) -> None:
         optim_index: int = candidate._meta["optim_index"]
```
Comment (jrapin) [inline nevergrad/optimization/optimizerlib.py:905]:
```suggestion
```

**Review by jrapin — APPROVED** [review]
Please remove the unnecessary line

**Comment by jrapin** [comment]
Also merge master and when it's done I'll try to figure out what happens with DE
It may have weird effects on optimizers though, so maybe it should be an option?

**Comment by jrapin** [comment]
@teytaud this is ready to merge if you want it to be merged, I expect this should *probably* be better, but not 100% sure (especially for DE, since it may lose diversity)

**Comment by teytaud** [comment]
@jrapin   I rerun test with the most recent master and I merge (way too late).

**Pull request merged**